src/Features/Core/Portable/Completion/Providers/Snippets/SnippetCompletionItem.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers.Snippets
+{
+    internal class SnippetCompletionItem
+    {
+        public static CompletionItem Create(
+            string displayText,
+            string displayTextSuffix,
+            TextSpan span,
+            Glyph glyph)
+        {
+            var props = ImmutableDictionary<string, string>.Empty
```

Comment:
I understand that `CompletionItem` is a public API, but with an increasing amount of diverse completion providers I think it might be time for us to consider extracting a base class for item and let each provider define their own subtypes, instead of using the cumbersome and inefficient property bag.

@CyrusNajmabadi thoughts?

Reply:
i don't see how that could work given that we'd have no idea how to do thigns like serialize completion items across processes or anything like that.  inheritance on a data-object allows people to add behaviors which we cannot capture.  The design of completion is *very* intentionally a split between behavior (in teh Service/Providers) and the data (the completion item).  The data needs to be pure data so that we can always handle it no matter how we split up where logic runs.